osu.Game.Rulesets.Taiko/Mods/TaikoModAlternate.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModAlternate : ModAlternate<TaikoHitObject, TaikoAction>
+    {
+        [SettingSource("Playstyle", "Preferred Alternate Playstyle")]
+        public Bindable<Playstyle> Style { get; } = new Bindable<Playstyle>();
+
+        private TaikoAction? lastAction1;
+        private TaikoAction? lastAction2;
+        private TaikoAction lastAction;
+        private TaikoHitObject lastHitObject;
+        private double lastActionTime = 0;
+        private const double strong_hit_window = 30;
```

Comment:
Where does this number come from?

Reply:
Defined here. It won't play well if ScoreV2 is properly implemented but I'm sure that it'll be a retrievable if it is.
https://github.com/ppy/osu/blob/860f10ceee2d1b18fd1578d2ad190499dc140394/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHit.cs#L208-L212